### PR TITLE
Parse memo field for PreStxOp

### LINKED ISSUE #3530: Add missing memo field for burnchain operations
The  `PreStxOp`, `StackStxOp` and `DelegateStxOp` burnchain operations are using less than 80 bytes of the OP_RETURN data. We would like to save the remaining bytes in these cases as a `memo: Vec<u8>` field which can be used for in-band signaling between miners.

This issue originates from a PR comment: [link](https://github.com/stacks-network/stacks-blockchain/pull/3486#issuecomment-1402705145).

### PR DESCRIPTION
**Pull request by pavitthrap**
### Description
Parsing memo field for the PreStxOp

### Applicable issues
- fixes #3530

### CODE REVIEW

File: src/chainstate/burn/operations/stack_stx.rs [hunk 1]
```diff
@@ -116,6 +117,9 @@ impl PreStxOp {
             return Err(op_error::InvalidInput);
         };
 
+        let mut memo = tx.data();
+        memo.truncate(77);
```
Comment (jcnelson) [inline src/chainstate/burn/operations/stack_stx.rs:121]: Is there a reason we're limiting it to 77 bytes?

**Review by jcnelson — COMMENTED** [review]
I don't think this is backwards-compatible with the existing chainstate -- previously-stored `PreStxOp`s won't be parseable with this new structure.

Is this change intended to be part of a point release, or is it part of the sBTC hard fork?  Because if it's the former, then I would highly recommend making it backwards compatible. If not, then no need to worry.

**Comment by pavitthrap** [comment]
> I don't think this is backwards-compatible with the existing chainstate -- previously-stored `PreStxOp`s won't be parseable with this new structure.
> 
> Is this change intended to be part of a point release, or is it part of the sBTC hard fork? Because if it's the former, then I would highly recommend making it backwards compatible. If not, then no need to worry.

Hmm.. PreStxOp's are actually not stored anywhere, so I'm not sure what exactly you're pointing to. 

Also, this is intended to be rolled into a hard fork, not a point release.

**Comment by jcnelson** [comment]
> Hmm.. PreStxOp's are actually not stored anywhere, so I'm not sure what exactly you're pointing to.

Ah, yes, you're right.  Was thinking of the other burnchain ops when I wrote that.

**Review by jcnelson — APPROVED** [review]

**Comment by wileyj** [comment]
ping @kantai @obycode

File: src/chainstate/burn/operations/stack_stx.rs (hunk 1 above)
Comment (obycode) [reply src/chainstate/burn/operations/stack_stx.rs:121]: Yes, why 77? Should it be 34, like the `stx-transfer-memo?`?

Comment (obycode) [reply src/chainstate/burn/operations/stack_stx.rs:121]: Oh, I see the stx transfer burnchain op allows up to 61 bytes, so the size is not consistent.

Comment (obycode) [reply src/chainstate/burn/operations/stack_stx.rs:121]: Ah, it's 77 because it's just 2 magic bytes, 1 byte for the op, and the rest are memo.

**Review by obycode — APPROVED** [review]
LGTM

**Pull request merged**